pkg/controller/job/job_controller.go

```diff
@@ -742,13 +749,19 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (forget bool, rEr
 	// Job first start. Set StartTime and start the ActiveDeadlineSeconds timer
 	// only if the job is not in the suspended state.
 	if job.Status.StartTime == nil && !jobSuspended(&job) {
-		now := metav1.Now()
-		job.Status.StartTime = &now
-		// enqueue a sync to check if job past ActiveDeadlineSeconds
-		if job.Spec.ActiveDeadlineSeconds != nil {
-			klog.V(4).Infof("Job %s has ActiveDeadlineSeconds will sync after %d seconds",
-				key, *job.Spec.ActiveDeadlineSeconds)
-			jm.queue.AddAfter(key, time.Duration(*job.Spec.ActiveDeadlineSeconds)*time.Second)
+		hasSetStartTime, startTime := jm.hasJobStartTimeSet(key)
```

Comment:
What if someone deletes an existing job and then creates a new job with the same namespace/name immediately?

The `jm.removeJobStartTime(key)` above may not be called because of the quick recreate, and here it will get the wrong startTime of the previous job.